integration-tests/tests/src/tests/sync/client-reset.ts

```diff
@@ -53,18 +54,16 @@ function getPartitionValue() {
   return new BSON.UUID().toHexString();
 }
 
-async function triggerClientReset(app: App, user: User): Promise<void> {
-  const maxAttempts = 5;
-  let deleted = false;
-  let count = maxAttempts;
-  while (count > 0) {
-    deleted = (await user.functions.triggerClientReset(app.id, user.id)) as boolean;
-    if (deleted) {
-      return;
-    }
-    count--;
+async function triggerClientReset(app: App, syncSession: Realm.App.Sync.Session): Promise<void> {
+  const { fileIdent } = syncSession as unknown as Record<string, unknown>;
+  if (typeof fileIdent !== "bigint") {
+    throw new Error("Expected the internal file ident");
   }
-  throw new Error(`Cannot trigger client reset in ${maxAttempts} attempts`);
+  await baasAdminClient.ensureLogIn();
+  const { _id } = await baasAdminClient.getAppByClientAppId(app.id);
+  syncSession.pause();
```

Comment:
Do we need a connection listener to ensure it is safe to call `forceSyncReset()`?

Reply:
Safe in what way? That we have access to BaaS from the test suite or perhaps that the Realm is ready to get the reset? (for the latter we are doing a `await syncSession.uploadAllLocalChanges()` before calling `triggerClientReset`).

Reply:
`pause()` will not immediately pause the connection, and I wonder if we need to wait for it before calling `forceSyncReset()`?

Reply:
Ah 🤔 I honestly don't know .. I think it's safe to assume `resume` will internally block for anything async that `pause` started on the sync client thread.